providers/storage/mongodocstore.js

```diff
@@ -36,7 +36,7 @@ class MongoDocStore {
     if (cached) {
       return Q(cached.document);
     }
-    return this.db.collection(type).findOne({ '_metadata.url': url }).then(value => {
+    return this.db.collection(type).findOne({ '$or': [{ '_metadata.url': url }, { '_metadata.links.self.href': url }] }).then(value => {
```

Comment:
why are you looking at the links?  that href should be the urn.

If you do want to do the or here and test for both, you likely need to add the same code in etag()